src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -97,7 +102,7 @@ private void initialize(NativeMemoryCacheManagerDto nativeMemoryCacheDTO) {
         }
 
         cacheCapacityReached = new AtomicBoolean(false);
-
+        accessRecencyQueue = new ConcurrentLinkedDeque<>();
```

Comment:
So - why cant we rely on guava cache to do this? I forget some of the details around how size based evictions take place.

Reply:
https://github.com/opensearch-project/k-NN/pull/2015#discussion_r1742382795